gulpfile.js

```diff
@@ -0,0 +1,40 @@
+/* eslint-env node */
+'use strict';
+
+const fs = require('fs');
+const fse = require('fs-extra');
+const gulp = require('gulp');
+const file = require('gulp-file');
+const path = require('path');
+const tar = require('tar-fs');
+const stagingWorkflow = require('bin-minify').stagingWorkflow;
+
+const {
+    GIT_BIN_PATH,
+    MIN_PACK_PATH,
+    MIN_PACK_FILENAME,
+} = require('./config');
+
+const GIT_FILENAME = 'git-2.13.5.tar';
+
+gulp.task('default', async (done) => {
```

Comment:
Why do we need gulp here? Could we make this a simple node script instead? Seems a little like overkill.

Reply:
We could migrate it to its own script. That might require some work around error handling, which gulp is handling for us now.

As this is being invoked by `npm` (see `package.json` -> `scripts` -> `prepublishOnly`) error handling would need to work nicely there as well, so `npm publish` does not happen when `prepublishOnly` fails.

I am willing to give it a try or we could leave it as a future enhancement and release the current version sooner rather than later. Thoughts?

Reply:
I do think it would be better to do this as a standalone script, and drop all those deps. If it's a lot of work it's ok, but it would be better if you've got time.

I'd be surprised if gulp is actually giving you much error handling here, and it shouldn't be necessary anyway - any stream that fires an `error` event without a specific handler will already crash the node process with a failing exit code immediately, and that should be sufficient to block publishing.

Reply:
done, `stageGit.js` worked from within `npm run prepublishOnly`

Reply:
And now I remember why I was using `gulp`...

It was to bypass issues around [`filecompare`](https://www.npmjs.com/package/filecompare) running on Node 6 & 7...

I do not know what gulp is doing that fixes it but without it Travis CI Node [6](https://travis-ci.org/pimterry/lambda-git/jobs/462040385) & [7](https://travis-ci.org/pimterry/lambda-git/jobs/462040386) tests fail with error:
```
/home/node/app/node_modules/filecompare/lib/index.js:3
const fsread = require('util').promisify(fs.read);
                                ^
 
TypeError: require(...).promisify is not a function
```

I am open to suggestions here but if we do not find a quick fix I vote for keeping `gulp` and moving on...